code/modules/antagonists/changeling/powers/mutations.dm

```diff
@@ -213,6 +213,18 @@
 	throw_range = 0
 	throw_speed = 0
 	var/datum/action/changeling/weapon/parent_action
+	var/hit_something = FALSE //Used for deleting gun after hitting something
+	var/shooting_right_now = FALSE //Used to track shooting to prevent deleting mid shot
```

Comment:
```suggestion
	/// Used for deleting gun after hitting something
	var/hit_something = FALSE
	/// True if we're shooting our shot -- used to track shooting to prevent deleting mid shot
	var/shooting_right_now = FALSE 
```